app/controllers/confirmation_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ConfirmationController < ApplicationController
+  rescue_from ActiveRecord::RecordNotUnique, with: :error_not_unique
+
+  # This populates instance variables used by the new confirmation form.
+  def new
+    @term = Term.find(params[:term_id])
+    @categories = Category.all
+    @confirmation = Confirmation.new
+  end
+
+  # This receives the submission from the new confirmation form, creating the needed record with the help of various
+  # private methods.
+  def create
+    confirmation = Confirmation.new({
+                                      term_id: params[:term_id],
+                                      user: current_user
+                                    })
+    set_category(confirmation, params[:confirmation][:category])
+    feedback_for(confirmation.save)
+    redirect_to terms_unconfirmed_path
+  end
+
+  private
+
+  # This method takes the result of the confirmation.save directive above and sets an appropriate flash message.
+  #
+  # The final else clause is likely to be difficult to provoke, but I have trouble building an if conditional without
+  # either a default value or an else clause.
+  def feedback_for(result)
+    if result == true && params[:confirmation][:category] == 'flag'
+      flash[:success] = 'Term flagged for review'
+    elsif result == true
+      flash[:success] = "Term confirmed as #{Category.find_by(id: params[:confirmation][:category]).name}"
+    else
+      flash[:error] = 'Unable to finish confirming this term. Please try again, or try a different term.'
+    end
+  end
+
+  # This method catches the RecordNotUnique error in case a duplicate confirmation is ever submitted, and shows an
```

Comment:
Are we sure we don't want a user to be able to "fix" a previous confirmation? I'm not sure if this would prevent that (maybe not?).

The use case I have in mind is to in the future allow for someone to click on a category and then go "oops I clicked wrong" and be able to update their confirmation (or delete it). I am fairly certain this won't prevent that, but want to check with you to make sure we both think this is just preventing _multiple_ confirmations but not preventing us from adding the ability to review recent confirmations and changing/removing them (in the future).

Reply:
Editing an existing record in this way shouldn't trip the uniqueness requirement, AFAICT - the constraint is on the pair of Term and User, and any edit would only affect the category value.

I agree that this use case would be useful, and also that it is future work. For now, I don't think there's anything here that would preclude that happening.